src/modules/product-category/dto/create-product-category.dto.ts

```diff
@@ -0,0 +1,39 @@
+import { IsNotEmpty, IsString, IsOptional, IsUUID } from 'class-validator';
+import { ApiProperty } from '@nestjs/swagger';
+
+export class CreateProductCategoryDto {
+  @ApiProperty({
+    description: 'Name of the product category',
+    example: 'Electronics',
+  })
+  @IsNotEmpty()
+  @IsString()
+  name: string;
+
+  @ApiProperty({
+    description: 'Description of the product category',
+    example: 'Electronic devices and accessories',
+    required: false,
+  })
+  @IsNotEmpty()
+  @IsString()
+  description?: string;
+
+  @ApiProperty({
+    description: 'Slug for the product category',
+    example: 'electronics',
+    required: false,
+  })
+  @IsNotEmpty()
+  @IsString()
+  slug?: string;
```

Comment:
If `description` is optional but, when provided, must be a non-empty string, won't using `@IsNotEmpty()` and `@IsString()` together not cause issues if description is not provided?

Please properly test all edge cases as well as for `slug` :)

Reply:
I get your point on this.
Removed @IsNotEmpty for both description and slog.